debugcc.c

```diff
@@ -191,7 +175,41 @@ static void measure(const struct measure_clk *clk)
 	if (clk->fixed_div)
 		raw_count_full *= clk->fixed_div;
 
-	printf("%50s: %fMHz (%ldHz)\n", clk->name, raw_count_full / 1000000.0, raw_count_full);
+
+	return raw_count_full;
+}
+
+static void measure(const struct measure_clk *clk)
+{
+	unsigned long clk_rate;
+	struct debug_mux *gcc = clk->primary;
+
+	if (!leaf_enabled(gcc, clk->leaf)) {
+		printf("%50s: skipping\n", clk->name);
+		return;
+	}
+
+	if (clk->leaf)
+		mux_prepare_enable(clk->leaf, clk->leaf_mux);
+
+	mux_prepare_enable(clk->primary, clk->mux);
+
+	if (clk->leaf && clk->leaf->measure)
+		clk_rate = clk->leaf->measure(clk);
+	else
+		clk_rate = measure_default(clk);
+
+	mux_disable(clk->primary);
```

Comment:
In hindsight I think Konrad's MCCC code wants to use the `->leaf` member to store some data about the MCCC block (register, `period_offset`) but we don't want to prepare nor enable the mux since there's nothing there. Unless every platform implements its own `measure_mccc` struct where this data is hardcoded.